src/cryptonote_core/blockchain.cpp

```diff
@@ -2123,6 +2123,18 @@ bool Blockchain::create_block_template_internal(
             return false;
         }
 
+        eth::L2Tracker::L2Heights l2_heights = m_l2_tracker->get_l2_heights();
+        if (l2_heights.synced < b.l2_height) {  // Haven't synced the logs for this height yet
+            log::error(
+                    logcat,
+                    "Block template uses L2 height {} which is not synced in the L2 tracker yet. "
+                    "The L2 tracker's current latest/synced height is {}/{}",
+                    b.l2_height,
+                    l2_heights.latest,
+                    l2_heights.synced);
+            return false;
+        }
```

Comment:
Better to move this a few lines, either to above `auto actual_reward` or below the `b._height = height;`: right now it's splitting the reward rate value from the reward rate clamping just below.